### PR TITLE
Add first pass of smart shopping ad

### PR DESCRIPTION
**Pull request by samphippen**

### CODE REVIEW

File: examples/shopping_ads/add_smart_shopping_ad.rb [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    campaign_resource_name,
+  )
+
+  add_smart_shopping_ad(
+    client,
+    customer_id,
+    ad_group_resource_name,
+  )
+
+  if ceate_default_listing_group
```
Comment (mcloonan) [inline examples/shopping_ads/add_smart_shopping_ad.rb]: Typo. Can you make sure you test this code path?

File: examples/shopping_ads/add_smart_shopping_ad.rb [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      '--customer-id MERCHANT-CENTER-ACCOUNT-ID',
+      String,
+      'Merchant center account id'
+    ) do |v|
+      options[:merchant_center_account_id] = v
+    end
+
+
+    opts.on('-l', '--create-default-listing-group CREATE-DEFAULT-LISTING-GROUP', String, 'create default listing group') do |v|
+      options[:create_default_listing_group] = if v == "true"
```
Comment (mcloonan) [inline examples/shopping_ads/add_smart_shopping_ad.rb]: WDYT of the convention established here for boolean command line flags: https://github.com/googleads/google-ads-ruby/blob/master/examples/shopping_ads/add_shopping_product_ad.rb#L215

Comment (samphippen) [reply examples/shopping_ads/add_smart_shopping_ad.rb]: that'll create a truthy value if the user passes the literal string "false", the canonical correct way to do this in ruby these days is:

https://github.com/rails/rails/blob/master/activemodel/lib/active_model/type/boolean.rb

but I'm not sure we want to pull in activemodel or vendor this code in just for code samples. WDYT?

Comment (mcloonan) [reply examples/shopping_ads/add_smart_shopping_ad.rb]: We definitely don't want to include any new dependencies for this. Can you update other examples that require true/false values to use whatever paradigm we go with here?

You can simplify this to: `options[:...] = v == "true"`

**Pull request closed without merging**